app/src/main/java/org/fossasia/openevent/adapters/LocationsListAdapter.java

```diff
@@ -105,18 +107,27 @@ public void moveItem(int fromPosition, int toPosition) {
         notifyItemMoved(fromPosition, toPosition);
     }
 
-    class Viewholder extends RecyclerView.ViewHolder {
+    public static interface AdapterOnClickHandler {
+        void onClick(int pos, Viewholder vh);
+    }
+
+    public static class Viewholder extends RecyclerView.ViewHolder implements View.OnClickListener {
         TextView name;
         TextView floor;
 
         public Viewholder(View itemView) {
             super(itemView);
             itemView.setClickable(true);
-
+            itemView.setOnClickListener(this);
             name = (TextView) itemView.findViewById(R.id.location_name);
             floor = (TextView) itemView.findViewById(R.id.location_floor);
 
         }
 
+        @Override
+        public void onClick(View view) {
+            clickHandler.onClick(getAdapterPosition(), this);
```

Comment:
so you will need to do 2 things:
* in the outer adapter you already cache the listener
* set the listener to the item view like `viewHolder.itemView.setListener(cachedListener)`
* Remove the `setOnClickListener` from viewholder constructor

That should be good.